### ISSUE TITLE
Fails with recursive types

### ISSUE THREAD

**Issue by jakearchibald** [dsherret/conditional-type-checks#18]
https://www.typescriptlang.org/play?#code/JYWwDg9gTgLgBAbzgQwM6oKawDRwJKoCiAHsgMbwC+cAZlBCHAORkQB2AJsDMO8gDYBaGAE8wGQWQAWGMgGtUTANwAoFWkywAPARLkYOgCIAhAGoDgHANIYRuPCfP9LNkQD43AChhQArhgBKJSA

```ts
import { assert, IsExact } from 'conditional-type-checks';

assert<IsExact<IDBValidKey, IDBValidKey>>(true);
```

"Type instantiation is excessively deep and possibly infinite."

This used to work, but something in Typescript changed.

**Comment by jakearchibald** [comment]
`DeepMakeRequiredForIsExact` is the cause

**Comment by dsherret** [comment]
The `IDBValidKey` type seems to have changed:

- TS 4.3.5: `string | number | Date | BufferSource | IDBArrayKey` - No error
- TS 4.4.4: `string | number | Date | BufferSource | IDBValidKey[]` - Error, since recursive

Fixed in 1.0.6 (already published)

**Comment by jakearchibald** [comment]
Thank you!
